src/main/java/org/apache/sling/xss/impl/xml/Regexp.java

```diff
@@ -27,12 +27,19 @@ public class Regexp {
     private String name;
     private String value;
 
+    private Pattern pattern;
+
     @JsonCreator
     public Regexp(@JacksonXmlProperty(localName = "name", isAttribute = true) String name,
             @JacksonXmlProperty(localName = "value", isAttribute = true) String regexp) {
 
         this.name = name;
         this.value = regexp;
+
+        if (regexp != null) {
```

Comment:
This check is a change in behavior, because now ``getPattern()`` can return ``null``, which was not possible before.
(``Pattern.compile()`` is not really descriptive when it comes to the handling of a ``null`` regexp string, but I would assume that it throws a ``PatternSyntaxException``.)

Are we sure that regexp is always non-null?

Reply:
If I understand the config file format and code correctly, the value can be null 

https://github.com/apache/sling-org-apache-sling-xss/blob/fb8a71fedffb4da0706cb60328244759547fbec4/src/main/resources/SLING-INF/content/config.xml#L254-L260

Not sure why it does not bubble up as a failure though (before or after this change).

Reply:
From what I see this structure is parsed via [Attribute.java](https://github.com/apache/sling-org-apache-sling-xss/blob/fb8a71fedffb4da0706cb60328244759547fbec4/src/main/java/org/apache/sling/xss/impl/xml/Attribute.java#L49) and there ``null`` Patterns are filtered out ([line 63](https://github.com/apache/sling-org-apache-sling-xss/blob/fb8a71fedffb4da0706cb60328244759547fbec4/src/main/java/org/apache/sling/xss/impl/xml/Attribute.java#L63)).

There are other instances of regexp tags, where ``getPattern()`` is called without filtering for null results.

Reply:
Let's continue the discussion in https://issues.apache.org/jira/browse/SLING-12622